android/src/main/java/nl/kega/reactnativeemdk/BarcodeScannerThread.java

```diff
@@ -115,280 +117,269 @@ private void destroyScanner() {
-                            break;
-                            case "codabar":  
-                                this.scanner_config.decoderParams.codaBar.enabled = true;
-                            break;
-                            case "code11":  
-                                this.scanner_config.decoderParams.code11.enabled = true;
-                            break;
-                            case "code39":  
-                                this.scanner_config.decoderParams.code39.enabled = true;
-                            break;
-                            case "code93":  
-                                this.scanner_config.decoderParams.code93.enabled = true;
-                            break;
-                            case "code128":  
-                                this.scanner_config.decoderParams.code128.enabled = true;
-                            break;
-                            case "qrcode":  
-                                this.scanner_config.decoderParams.qrCode.enabled = true;
-                            break;
-                            case "dutchpostal":  
-                                this.scanner_config.decoderParams.dutchPostal.enabled = true;
-                            break;
-                            
-                        }
-                        
-                    }
-                }else{
-                    this.scanner_config.decoderParams.ean8.enabled = true;
-                    this.scanner_config.decoderParams.ean13.enabled = true;
-                    this.scanner_config.decoderParams.codaBar.enabled = true;
-                    this.scanner_config.decoderParams.code11.enabled = true;
-                    this.scanner_config.decoderParams.code39.enabled = true;
-                    this.scanner_config.decoderParams.code93.enabled = true;
-                    this.scanner_config.decoderParams.code128.enabled = true;
-                    this.scanner_config.decoderParams.qrCode.enabled = true;
-                    this.scanner_config.decoderParams.dutchPostal.enabled = true;
-                }
-
-                this.scanner_config.scanParams.decodeHapticFeedback = true;
-
-                this.scanner.setConfig(this.scanner_config);
-
+                this.scanner.setConfig(configureScanner(scanner));
                 this.scanner.read();
-
             }
-
         } catch (ScannerException e) {
-            Log.e("[BarcodeScanner]", "Read error: " + e);
+            Log.e(TAG, "Read error: " + e);
         } catch (NullPointerException e) {
-        
+            Log.e(TAG, "Read null pointer error: " + e);
         }
-
     }
 
+    /**
+     * Sets TriggerType to SOFT_ALWAYS and calls Scanner.read().
+     * This method should be used when only using a software button to
+     * trigger the scanner.
+     *
+     * @param condig Scanner configuration passed from React Native app.

```

Comment:
What are the possible values for the configuration and is the React Native app the only caller type for this method?

Reply:
https://github.com/lensrentals/react-native-emdk/pull/2/files#r287499811